Assets/FlyingEnemyAI.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using UnityEngine;
+
+public class FlyingEnemyAI : MonoBehaviour
+{
+    [SerializeField] private float health = 100f;
+
+    public float speed = 2f;
+    public GameObject projectilePrefab;
+    public float fireRate = 1f;
+    private float fireTimer = 0f;
+    private Transform playerTransform;
+
+    void Start()
+    {
+        playerTransform = GameObject.FindWithTag("Player").transform;
+    }
+
+    void Update()
+    {
+        // Move towards the player on the x-axis only
+        Vector2 newPos = Vector2.MoveTowards(transform.position, new Vector2(playerTransform.position.x, transform.position.y), speed * Time.deltaTime);
+        transform.position = new Vector3(newPos.x, transform.position.y, transform.position.z);
+
+        // Fire projectile towards player's position
+        fireTimer += Time.deltaTime;
+        if (fireTimer >= fireRate)
+        {
+            fireTimer = 0f;
+            Instantiate(projectilePrefab, transform.position, Quaternion.identity);
+        }
```

Comment:
Maybe this code (or a portion of it) could also go into the common interface that all AI will use?